File: Graph/ScriptTemplate/And/CGAnd.js
/**
 * @file CGAnd.js
 * @author
 * @date 2021/8/15
 * @brief CGAnd.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGAnd extends BaseNode {
  constructor() {
    super();
  }

  getOutput(index) {
    return this.inputs[0]() && this.inputs[1]();
  }
}

exports.CGAnd = CGAnd;

File: Graph/ScriptTemplate/And/CGAnd.lua
local CGAnd = CGAnd or {}
CGAnd.__index = CGAnd

function CGAnd.new()
    local self = setmetatable({}, CGAnd)
    self.inputs = {}
    return self
end

function CGAnd:setInput(index, func)
    self.inputs[index] = func
end

function CGAnd:getOutput(index)
    return self.inputs[0]() and self.inputs[1]()
end

return CGAnd
